packages/integrations/netlify/src/netlify-functions.ts

```diff
@@ -8,7 +8,11 @@ polyfill(globalThis, {
 });
 
 export interface Args {
-	builders?: boolean;
+	builders?:
+		| boolean
+		| {
+				ttl?: number | ((event: HandlerEvent) => number | undefined);
```

Comment:
Unfortunately this kind of API won't work. `Args` gets serialized and included in the build. Your astro.config.mjs does not get bundled with the build. So only plain objects/primitives can be passed into the actual integration. Functions won't work.

How we usually fix this is by instead having the code be defined in a separate module and then building that module as part of the build. Something: `builders: 'path/to/implementation.ts'`. It's not as nice of an API but it prevents us from needing to build/bundle the config file

Reply:
@matthewp Oh wow, I wasn't aware of that. Does this only apply to adapter configurations? After all we can use Vite plugins in `astro.config` that again are (partly) functions: https://docs.astro.build/en/reference/configuration-reference/#vite – on the other hand I can't find any example of passing a path to .ts or .js file. If you know of any other usage, that would be very helpful indeed.

Reply:
We don't build the config, so any functions that need to exist in production cannot be provided via a config option.

As for examples, all of the renderers work this way, see for example: https://github.com/withastro/astro/blob/e58a92527f54cb29d3515d544fad833a5ce1061d/packages/integrations/vue/src/index.ts#L13-L17

I'd suggest a similar pattern if you want to allow the user to customize runtime values.